webapp/src/ts/modules/tasks/tasks.component.ts

```diff
@@ -116,52 +128,86 @@ export class TasksComponent implements OnInit, OnDestroy {
+  private async hydrateEmissions(taskDocs) {
     return taskDocs.map(taskDoc => {
       const emission = { ...taskDoc.emission };
       const dueDate = moment(emission.dueDate, 'YYYY-MM-DD');
       emission.date = new Date(dueDate.valueOf());
       emission.overdue = dueDate.isBefore(moment());
       emission.owner = taskDoc.owner;
-
       return emission;
     });
   }
 
-  private refreshTasks() {
-    const telemetryData:any = {
-      start: Date.now(),
-    };
-
-    return this.rulesEngineService
-      .isEnabled()
-      .then(isEnabled => {
-        this.tasksDisabled = !isEnabled;
-        return isEnabled ? this.rulesEngineService.fetchTaskDocsForAllContacts() : [];
-      })
-      .then(taskDocs => {
-        this.hasTasks = taskDocs.length > 0;
-        this.loading = false;
-        this.tasksActions.setTasksList(this.hydrateEmissions(taskDocs));
-        if (!this.tasksLoaded) {
-          this.tasksActions.setTasksLoaded(true);
-        }
-
-        telemetryData.end = Date.now();
-        const telemetryEntryName = !this.tasksLoaded ? `tasks:load`: `tasks:refresh`;
-        this.telemetryService.record(telemetryEntryName, telemetryData.end - telemetryData.start);
-      })
-      .catch(err => {
-        console.error('Error getting tasks for all contacts', err);
-
-        this.error = true;
-        this.loading = false;
-        this.hasTasks = false;
-        this.tasksActions.setTasksList([]);
-      });
+  async refreshTasks() {
+    try {
+      const telemetryData:any = {
+        start: Date.now(),
+      };
+
+      const isEnabled = await this.rulesEngineService.isEnabled();
+      this.tasksDisabled = !isEnabled;
+      const taskDocs = isEnabled ? await this.rulesEngineService.fetchTaskDocsForAllContacts() : [];
+
+      this.hasTasks = taskDocs.length > 0;
+      this.loading = false;
+
+      const hydratedTasks = await this.hydrateEmissions(taskDocs) || [];
+      const subjects = await this.getLineagesFromTaskDocs(hydratedTasks);
+
+      if(subjects) {
+        hydratedTasks.forEach(task => {
+          const lineage = this.getTaskLineage(subjects, task);
+          task.lineage = this.currentLevel ? this.removeCurrentLineage(lineage) : lineage;
+        });
+      }
+
+      this.tasksActions.setTasksList(hydratedTasks);
+
+      if (!this.tasksLoaded) {
+        this.tasksActions.setTasksLoaded(true);
+      }
+
+      telemetryData.end = Date.now();
+      const telemetryEntryName = !this.tasksLoaded ? `tasks:load`: `tasks:refresh`;
+      this.telemetryService.record(telemetryEntryName, telemetryData.end - telemetryData.start);
+       
+    } catch (exception) {
+      console.error('Error getting tasks for all contacts', exception);
+      this.error = true;
+      this.loading = false;
+      this.hasTasks = false;
+      this.tasksActions.setTasksList([]);
+    }
   }
 
   listTrackBy(index, task) {
     return task?._id;
   }
+
+  getCurrentLineageLevel(){
```

Comment:
Since it's not used in the DOM
```suggestion
  private getCurrentLineageLevel() {
```